src/server/dfly_main.cc

```diff
@@ -438,6 +439,114 @@ bool ShouldUseEpollAPI(const base::sys::KernelVersion& kver) {
   return true;
 }
 
+string get_cgroup_path(void) {
+  // Begin by reading /proc/self/cgroup
+
+  auto cg = io::ReadFileToString("/proc/self/cgroup");
+  CHECK(cg.has_value());
+
+  // Here, we assume that CGroup v1 is being used. This
+  // is quite likely, as CGroup v1 was introduced back in 2015.
+
+  // strip 0::<path> into just <path>, and newline.
+  auto cgv = std::move(cg.value());
+  return cgv.substr(3, cgv.length() - 3 - 1);
+  /**
+   * -3: we are skipping the first three characters
+   * -1: we discard the last character (a newline)
+   */
+}
+
+const auto cgroup = "/sys/fs/cgroup/" + get_cgroup_path();
```

Comment:
Please avoid using non-POD global variables as much as possible. It may create weird bugs since they are initialized before `main(..)` starts.